src/dev/i18n/extract_default_translations.js

```diff
@@ -38,15 +42,55 @@ function addMessageToMap(targetMap, key, value) {
   targetMap.set(key, value);
 }
 
-export async function extractDefaultTranslations(inputPath) {
+function filterPaths(inputPaths) {
+  const availablePaths = Object.values(paths);
+  const pathsForExtraction = new Set();
+
+  for (const inputPath of inputPaths) {
+    const normalizedPath = normalize(path.relative('.', inputPath));
+
+    if (normalizedPath) {
```

Comment:
__suggestion:__ Am I missing something or we can collapse that into the following?
```js
for (const inputPath of inputPaths) {
    const normalizedPath = normalizePath(inputPath);

    // If input path is the sub path of or equal to any available path, include it.
    if (availablePaths.some(path => normalizedPath.startsWith(`${path}/`) || path === normalizedPath)) {
      pathsForExtraction.add(normalizedPath);
    } else {
      // Otherwise go through all available paths and see if any of them is the sub
      // path of the input path (empty normalized path corresponds to root or above).
      availablePaths
        .filter(path => !normalizedPath || path.startsWith(`${normalizedPath}/`))
        .forEach(path => pathsForExtraction.add(path));
    }
  }
```

I added comments while trying to figure out what this code does exactly (I know you explained that already, but we need to have comments to others).